provider/pkg/provider/provider_intrinsics.go

```diff
@@ -73,22 +73,39 @@ func (p *cfnProvider) cidr(ctx context.Context, inputs resource.PropertyMap) (re
 	}
 
 	subnets := make([]resource.PropertyValue, int(count.NumberValue()))
-	subnets[0] = resource.NewStringProperty(network.String())
+	startPrefixLen, _ := network.Mask.Size()
+
+	prefixLen := int(cidrBits.NumberValue()) + startPrefixLen
+	if prefixLen > len(network.IP)*8 {
+		protocol := "IP"
+		switch len(network.IP) * 8 {
+		case 32:
+			protocol = "IPv4"
+		case 128:
+			protocol = "IPv6"
+		}
+		return nil, fmt.Errorf("cidrBits %d would extend prefix to %d bits, which is too long for an %s address", int(cidrBits.NumberValue()), prefixLen, protocol)
+	}
 
-	prefixLen := int(cidrBits.NumberValue())
-	for i := 1; i < len(subnets)-1; i++ {
-		subnet, ok := gocidr.NextSubnet(network, prefixLen)
-		if !ok {
-			return nil, fmt.Errorf("could not create %d subnets", len(subnets))
+	current, _ := gocidr.PreviousSubnet(network, prefixLen)
```

Comment:
What's the underscore here? Should we `contract.Assertf(ok, "PreviousSubnet lookup failed")` or inspect it?